### PR TITLE
Adds documentation for multiple dynamic credentials config support for the AzureAD provider

### PR DESCRIPTION
**Pull request by mpminardi**
### What
Adds documentation for AzureAD support for multiple provider configurations

### Why
Support for multiple provider configuration for the AzureAD provider was added in the most recently provider release (v2.43.0).

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] (N/A) Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [x] (N/A) API documentation and the API Changelog have been updated. 
- [x] (N/A) Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] (N/A) Pages with related content are updated and link to this content when appropriate.
- [x] (N/A) Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] (N/A) New pages have metadata (page name and description) at the top.
- [x] (N/A) New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] (N/A) UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

#### External Links
- [JIRA](https://hashicorp.atlassian.net/browse/TF-9426)

### CODE REVIEW

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/azure-configuration.mdx [hunk 1]
```diff
@@ -115,6 +121,37 @@ provider "azurerm" {
...
+  tenant_id            = "20000000-0000-0000-0000-000000000000"
+}
+```
+
+##### AzureAD Provider
+
+```hcl
+provider "azuread" {
+  features {}
+  // use_cli should be set to false yield more accurate error messages on auth failure.
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/azure-configuration.mdx:141]: Quick suggestion (same as above!):
```suggestion
  // use_cli should be set to false to yield more accurate error messages on auth failure.
```

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/azure-configuration.mdx [hunk 2]
```diff
@@ -115,6 +121,37 @@ provider "azurerm" {
...
+  use_oidc             = true
+  client_id_file_path  = var.tfc_azure_dynamic_credentials.default.client_id_file_path
+  oidc_token_file_path = var.tfc_azure_dynamic_credentials.default.oidc_token_file_path
+  subscription_id      = "00000000-0000-0000-0000-000000000000"
+  tenant_id            = "10000000-0000-0000-0000-000000000000"
+}
+
+provider "azuread" {
+  features {}
+  // use_cli should be set to false yield more accurate error messages on auth failure.
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/azure-configuration.mdx:153]: Quick suggestion (same as above!):
```suggestion
  // use_cli should be set to false to yield more accurate error messages on auth failure.
```

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/vault-backed/azure-configuration.mdx [hunk 3]
```diff
@@ -88,9 +90,13 @@ variable "tfc_vault_backed_azure_dynamic_credentials" {
 
 #### Example Usage
 
+##### AzureRM Provider
+
 ```hcl
 provider "azurerm" {
   features {}
+  // use_cli should be set to false yield more accurate error messages on auth failure.
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/vault-backed/azure-configuration.mdx:98]: Quick suggestion (same as above!):
```suggestion
  // use_cli should be set to false to yield more accurate error messages on auth failure.
```

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/vault-backed/azure-configuration.mdx [hunk 4]
```diff
@@ -99,10 +105,37 @@ provider "azurerm" {
 
 provider "azurerm" {
   features {}
+  // use_cli should be set to false yield more accurate error messages on auth failure.
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/vault-backed/azure-configuration.mdx:108]: Quick suggestion (same as above!):
```suggestion
  // use_cli should be set to false to yield more accurate error messages on auth failure.
```

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/vault-backed/azure-configuration.mdx [hunk 5]
```diff
@@ -99,10 +105,37 @@ provider "azurerm" {
...
   tenant_id               = "20000000-0000-0000-0000-000000000000"
 }
 ```
+
+##### AzureAD Provider
+
+```hcl
+provider "azuread" {
+  features {}
+  // use_cli should be set to false yield more accurate error messages on auth failure.
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/vault-backed/azure-configuration.mdx:123]: Quick suggestion (same as above!):
```suggestion
  // use_cli should be set to false to yield more accurate error messages on auth failure.
```

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/vault-backed/azure-configuration.mdx [hunk 6]
```diff
@@ -99,10 +105,37 @@ provider "azurerm" {
...
+  use_cli                 = false
+  client_id_file_path     = var.tfc_vault_backed_azure_dynamic_credentials.default.client_id_file_path
+  client_secret_file_path = var.tfc_vault_backed_azure_dynamic_credentials.default.client_secret_file_path
+  subscription_id         = "00000000-0000-0000-0000-000000000000"
+  tenant_id               = "10000000-0000-0000-0000-000000000000"
+}
+
+provider "azuread" {
+  features {}
+  // use_cli should be set to false yield more accurate error messages on auth failure.
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/workspaces/dynamic-provider-credentials/vault-backed/azure-configuration.mdx:133]: Quick suggestion (same as above!):
```suggestion
  // use_cli should be set to false to yield more accurate error messages on auth failure.
```

**Review by juliannatetreault — COMMENTED** [review]
Left some suggestions, otherwise, looks great! 🌟

File: website/docs/cloud-docs/workspaces/dynamic-provider-credentials/azure-configuration.mdx [hunk 7]
```diff
@@ -102,9 +104,13 @@ variable "tfc_azure_dynamic_credentials" {
 
 #### Example Usage
 
+##### AzureRM Provider
+
 ```hcl
 provider "azurerm" {
   features {}
+  // use_cli should be set to false yield more accurate error messages on auth failure.
```
[the remark this answers is not in the document]
Comment (mpminardi) [reply website/docs/cloud-docs/workspaces/dynamic-provider-credentials/azure-configuration.mdx]: Ah thanks for pointing these out! Should be fixed now

**Review by juliannatetreault — APPROVED** [review]
⭐

**Pull request merged**